profiles/api/api.py

```diff
@@ -62,7 +63,10 @@ def _get(profile_id: str) -> Response:
         except ProfileNotFound as exception:
             raise NotFound(str(exception)) from exception
 
-        response = make_response(json.dumps(profile, default=normalize))
+        if request.headers.get('x-consumer-groups') == 'view-restricted-profiles':
+            response = make_response(json.dumps(profile, default=normalize_restricted))
+        else:
+            response = make_response(json.dumps(profile, default=normalize))
```

Comment:
I'm not sure if the fields involved are appearing in `/profiles`, it may need to be checked there too

In general I've seen in other services that the header is handled for all APIs:
https://github.com/elifesciences/digests/blob/4b418049799a999193e5ed4eb80b7587a123b0d8/app/core/middleware.py#L22

Reply:
`/profiles` returns the results of calling `Profiles.list`. This in turn does not have a parameter to specify restricted data. This may raise the question of, if an ORCID user has their profile set with `trusted` permissions for their `name` and `id`, do we expected to see this in the Profiles database?

Reply:
I remember [handling of private names](https://github.com/elifesciences/profiles/pull/31) being introduced but not of `trusted` ones. If we go for the same behavior we would reject the user? Probably needs a check on existing profiles but I think it's difficult to correlate (if we don't store that information we will need to call ORCID with all the various tokens).
I also found the [original implementation](https://github.com/elifesciences/profiles/pull/58) of the restricted access if that helps with context.

Reply:
I've spoken with @thewilkybarkid and we think that it is only needed here for the time being. If there are more use cases then this can be extracted into middleware.